src/agent/src/image.rs

```diff
@@ -57,9 +57,15 @@ pub struct ImageService {
 
 impl ImageService {
     pub fn new() -> Self {
+        let mut image_client = ImageClient::new(PathBuf::from(KATA_IMAGE_WORK_DIR));
+        let images = HashMap::new();
+        if let Some(registry_auth) = &AGENT_CONFIG.image_registry_auth {
+            debug!(sl(), "Set registry auth file {:?}", registry_auth);
+            image_client.config.file_paths.auth_file = registry_auth.to_string_lossy().into();
```

Comment:
Hi @mkulke !

Reading [this line](https://github.com/confidential-containers/guest-components/blob/main/image-rs/src/image.rs#L222) in image-rs' `pull_image()` and image-rs' credential test https://github.com/confidential-containers/guest-components/blob/main/image-rs/tests/credential.rs#L38

Reply:
I've cherry-picked this and updated it in my version.